omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+        obstacles += [target_obstacle.get_obstacle(next(iter(obj.root_link.visual_meshes.values())).prim_path)]
+        n_spheres_total += [n_spheres]
+
+    if extra_links is not None:
+        extra_coll_paths = [geom.prim_path for link in extra_links for geom in link.collision_meshes.values()]
+        extra_obstacles = get_obstacles(
+            reference_prim_path=reference_prim_path,
+            only_substring=extra_coll_paths,
+        )
+        obstacles += [extra_obstacles.get_obstacle(coll_path) for coll_path in extra_coll_paths]
+        n_spheres_total += [n_extra_spheres] * len(extra_coll_paths)
+
+    return get_obstacles_sphere_representation(
+        obstacles=obstacles,
+        tensor_args=tensor_args,
+        n_spheres=n_spheres_total,
+        sphere_radius=sphere_radius,
+    )
+
+
+class CuRoboMotionGenerator:
+    """
+    Class for motion generator using CuRobo backend
+    """
+
+    def __init__(
+        self,
+        robot,
+        robot_cfg_path=None,
+        ee_link=None,
+        device="cuda:0",
+        motion_cfg_kwargs=None,
+        ignore_collision_paths=None,
+        batch_size=2,
+        debug=False,
+    ):
+        """
+        Args:
+            robot (BaseRobot): Robot for which to generate motion plans
+            robot_cfg_path (None or str): If specified, the path to the robot configuration to use. If None, will
+                try to use a pre-configured one directly from curobo based on the robot class of @robot
+            ee_link (None or str): If specified, the link name representing the end-effector to track. None defaults to
+                value already set in the config from @robot_cfg
+            device (str): Which device to use for curobo
+            motion_cfg_kwargs (None or dict): If specified, keyward arguments to pass to
+                MotionGenConfig.load_from_robot_config(...)
+            ignore_collision_paths (None or list): If specified, prim path(s) to prims that should be ignored
+                during collision checking
+            batch_size (int): Size of batches for computing trajectories. This must be FIXED
+            debug (bool): Whether to debug generation or not
+        """
+        # Only support one scene for now -- verify that this is the case
+        assert len(og.sim.scenes)
+
+        # Define arguments to pass to motion gen config
+        self._tensor_args = lazy.curobo.types.base.TensorDeviceType(device=th.device(device))
+        self.debug = debug
+
+        # Load robot config and make sure paths point correctly
+        robot_cfg_path = robot.curobo_path if robot_cfg_path is None else robot_cfg_path
+        robot_cfg = lazy.curobo.util_file.load_yaml(robot_cfg_path)["robot_cfg"]
+        robot_cfg["kinematics"]["external_asset_path"] = gm.ASSET_PATH
+        robot_cfg["kinematics"]["external_robot_configs_path"] = robot_cfg_path
```

Comment:
What's in the config file? If we're only using it to load back into a dict, can't we just generate that dict?

Reply:
So iiuc there is some code that can generate this file for each robot, but the code is slow enough that the file is better pre-generated?

Reply:
Well to be clear, we need to manually specify this file for each robot. I don't think it can be generated (well) programmatically